src/MongoDB.Bson/ObjectModel/Decimal128.cs

```diff
@@ -793,7 +793,7 @@ public static double ToDouble(Decimal128 d)
             {
                 // TODO: implement this more efficiently
                 var stringValue = d.ToString();
```

Comment:
I think you need to make ToString be culture invariant also.

Reply:
I take this back... I didn't realize that WE wrote this particular ToString method and OUR implementation is already culture invariant.